pkgs/test/lib/src/runner/browser/dom.dart

```diff
@@ -99,61 +161,99 @@ extension NodeExtension on Node {
 class EventTarget {}
 
 extension EventTargetExtension on EventTarget {
-  void addEventListener(String type, EventListener? listener,
+  @JS('addEventListener')
+  external JSVoid _addEventListener1(JSString type, JSFunction? listener);
+  @JS('addEventListener')
+  external JSVoid _addEventListener2(JSString type, JSFunction? listener,
+      JSBoolean useCapture);
```

Comment:
Are JS interop definitions allowed to use optional arguments?

Reply:
Yes, JS interop definitions can use optional arguments. And they have JS semantics, that is not-passed is `undefined`. However, this means that these definitions do not have Dart semantics, i.e. default values, and they also have inconsistent semantics for tearoffs. I'm hoping we disallow tearoffs entirely, and instead users who want to tearoff these methods should wrap them with Dart functions as I do in this CL.

We don't have that check just yet though, so this seems like a safer path for now. If you prefer, we can use optional methods, but it may not be worth the possible pain because this shim is going to go away in the near future to be replaced by `package:web` anyways, so it's your call.

Thoughts?